manabi/lock.py

```diff
@@ -94,3 +117,125 @@ def get_lock_list(self, path, *, include_root, include_children, token_only):
                 include_children=include_children,
                 token_only=token_only,
             )
+
+
+class PostgresDict(MutableMapping):
+    def __init__(self, connection: psycopg2.extensions.connection):
+        self._connection = connection
+
+    @staticmethod
+    def decode_lock(lock):
+        # Encoding is not needed since a string is also valid for wsgidav
+        if "owner" in lock:
+            owner = lock["owner"]
+            if isinstance(owner, bytes):
+                lock["owner"] = lock["owner"].decode("UTF-8")
+
+    def cursor(self) -> psycopg2.extensions.cursor:
+        return self._connection.cursor()
+
+    def cleanup(self):
+        cursor = self.cursor()
+        cursor.execute("DELETE FROM manabi_lock;")
+
+    def __len__(self):
+        cursor = self.cursor()
+        cursor.execute("SELECT count(*) FROM manabi_lock")
+        return int(cursor.fetchone())
```

Comment:
`fetchone()` returns a record, not a single value.. Should probably be 
```suggestion
        cursor.execute("SELECT count(*) FROM manabi_lock")
        return int(cursor.fetchone()[0])
```

Reply:
Thanks is fixed